File: DotNet/LeanFTSampleProject/LeanFtSampleProject/UnitTestClassBase.cs
using System;
using System.Diagnostics;
using System.IO;
using System.Linq;
using System.Threading;
using HP.LFT.Report;
using HP.LFT.UnitTesting;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace LeanFtSampleProject
{
    public abstract class UnitTestClassBase<TDerive> : UnitTestBase where TDerive : UnitTestClassBase<TDerive>, new()
    {
        private static readonly UnitTestClassBase<TDerive> Instance = new TDerive();
        private static TestContext _testContext;

        public TestContext TestContext
        {
            get { return _testContext; }
            set { _testContext = value; }
        }

        [ClassInitialize]
        public static void GlobalSetup(TestContext context)
        {
            _testContext = context;
            Instance.UnitTestBaseSetup();
        }

        private void UnitTestBaseSetup()
        {
            Instance.TestSuiteSetup();
        }

        [ClassCleanup]
        public static void GlobalTearDown()
        {
            Instance.TestSuiteTearDown();
            Instance.Reporter.GenerateReport();
        }

        [TestInitialize]
        public void BasicSetUp()
        {
            Instance.TestSetUp();
        }

        [TestCleanup]
        public void BasicTearDown()
        {
            Instance.TestTearDown();
        }

        protected override string GetTestName()
        {
            return TestContext.TestName;
        }

        protected override string GetClassName()
        {
            return _testContext.FullyQualifiedTestClassName;
        }

        protected override Status GetFrameworkTestResult()
        {
            switch (_testContext.CurrentTestOutcome)
            {
                case UnitTestOutcome.Error:
                case UnitTestOutcome.Failed:
                    return Status.Failed;
                case UnitTestOutcome.Passed:
                    return Status.Passed;
                case UnitTestOutcome.Inconclusive:
                case UnitTestOutcome.Unknown:
                    return Status.Warning;
                default:
                    return Status.Passed;
            }
        }


    }
}

File: Java/LeanFTSampleProject/src/unittesting/UnitTestClassBase.java
package unittesting;

import org.junit.After;
import org.junit.Before;
import org.junit.ClassRule;
import org.junit.Rule;
import org.junit.rules.TestWatcher;

import com.hp.lft.report.Status;
import com.hp.lft.unittesting.UnitTestBase;

public class UnitTestClassBase extends UnitTestBase {

    private static UnitTestClassBase instance;

    public static void globalSetup(Class<? extends UnitTestClassBase> testClass) throws Exception {
        instance = testClass.newInstance();
        instance.classSetup();      
    }

    @Before
    public void beforeTest() {
        testSetup();
    }

    @After
    public void afterTest() {
        testTearDown();
    }

    public static void globalTearDown() throws Exception {
        instance.classTearDown();
        getReporter().generateReport();
    }

    @ClassRule
    public static TestWatcher classWatcher = new TestWatcher() {
        @Override
        protected void starting(org.junit.runner.Description description) {
            className = description.getClassName();
        }
    };

    @Rule
    public TestWatcher watcher = new TestWatcher() {
        @Override
        protected void starting(org.junit.runner.Description description) {
            testName = description.getMethodName();
        }
        @Override
        protected void failed(Throwable e, org.junit.runner.Description description) {
            setStatus(Status.Failed);
        }
        
        @Override
        protected void succeeded(org.junit.runner.Description description) {
            setStatus(Status.Passed);
        }
    };

    @Override
    protected String getTestName() {
        return testName;
    }
    
    @Override
    protected String getClassName() {
        return className;
    }

    protected static String className;
    protected String testName;
    
}